biggraphite/plugins/carbon.py

```diff
@@ -146,10 +146,11 @@ def write(self, metric_name, datapoints):
         metric_name = self.encode(metric_name)
 
         # Get a Metric object from metric name.
-        metric = self.cache.get_metric(metric_name=metric_name, touch=True)
+        metric = self.cache.get_metric(metric_name=metric_name)
         if not metric:
             raise Exception('Could not find %s' % (metric_name))
 
+        self.accessor.touch_metric(metric)
```

Comment:
Add a comment here explaining why

Reply:
should we rename touch_metric then ?

Reply:
Not necessarily, just explaining that we call it here because we are writing and thus may want to update the timestamps might be enough. You can also just call touch_metric in insert_points()